render/sprite-draw.go

```diff
@@ -221,3 +230,19 @@ func ToggleFiltering() {
 
 	fmt.Println(message)
 }
+
+func TogglePixelSnap() {
+	switch spriteProgram {
+	case &spriteProgram1:
+		spriteProgram = &spriteProgram2
+		fmt.Println("PIXEL SNAPPING: snapping to a pixel")
+	case &spriteProgram2:
+		spriteProgram = &spriteProgram3
+		fmt.Println("PIXEL SNAPPING: snapping to half a pixel")
+	case &spriteProgram3:
+		spriteProgram = &spriteProgram1
+		fmt.Println("PIXEL SNAPPING: no snapping")
+	default:
+		log.Fatalf("Pixel snapping error")
+	}
```

Comment:
Storing a list and an index may be simpler than switch/casing pointers

Reply:
will change it